docs/pattern-matching/as.md

```diff
@@ -1,34 +1,48 @@
 # As Operator
 
-The `as` operator in Pony has two related uses. First, it provides a safe way to increase the specificity of an object's type (casting). Second, it gives the programmer a way to specify the type of the items in an array literal.
+The `as` operator in Pony has two related uses. First, it provides a safe way to increase the specificity of an object's type. Second, it gives the programmer a way to specify the type of the items in an array literal.
 
-## Safely converting to a more specific type (casting)
+## Expressing a different type of an object
 
-The `as` operator can be used to create a reference to an object with a more specific type than the given reference, if possible. This can be applied to types that are related through inheritance, as well as unions and intersections. This is done at runtime, and if it fails then an error is raised.
+In Pony, each object instance has a single concrete type and can fulfill one or more interfaces or traits. In short hand, we often refer to this as an instance having multiple types. `as` allows an object alias to be expressed as one of the other types of the object which is not implied by the alias type.
```

Comment:
I had a lot of trouble reading through this paragraph, even given the context I already have about this feature. It doesn't map very well to the mental model I personally have, in that I don't think about it as "an instance having multiple types".

Let me take a shot at a different way of expressing it. Note that this is intended to replace both this paragraph and the one below it, but GitHub won't let me do a suggestion on the set pair of lines because it thinks that the set contains a deleted line.

```suggestion
In Pony, each object is an instance of a single concrete type, which is the most specific type for that object. But the object can also be held as part of a "wider" abstract type such as an interface, trait, or type union which the concrete type is said to be a subtype of.

`as` (like `match`) allows a program to check the runtime type of a abstract-typed value to see whether or not the object matches a given type which is more specific. If it doesn't match the more specific type, then a runtime `error` is raised. For example:
```